src/main/java/org/fogbowcloud/manager/core/utils/ComputeInstanceConnectivityUtil.java

```diff
@@ -0,0 +1,46 @@
+package org.fogbowcloud.manager.core.utils;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.instances.ComputeOrderInstance;
+import org.fogbowcloud.manager.core.threads.FulfilledMonitor;
+
+import java.util.Map;
+
+/**
+ * Class to check SSH connectivity with an compute instance.
+ */
+public class ComputeInstanceConnectivityUtil {
+
+    private static final Logger LOGGER = Logger.getLogger(FulfilledMonitor.class);
+
+    private TunnelingServiceUtil tunnelingService;
+    private SshConnectivityUtil sshConnectivity;
+
+    public ComputeInstanceConnectivityUtil(TunnelingServiceUtil tunnelingService, SshConnectivityUtil sshConnectivity) {
+        this.tunnelingService = tunnelingService;
+        this.sshConnectivity = sshConnectivity;
+    }
+
+    public void setTunnelingServiceAddresses(Order order, ComputeOrderInstance computeOrderInstance) {
+        try {
+            Map<String, String> externalServiceAddresses = this.tunnelingService.getExternalServiceAddresses(order.getId());
+            if (externalServiceAddresses != null) {
+                computeOrderInstance.setExternalServiceAddresses(externalServiceAddresses);
+            }
+        } catch (Throwable e) {
+            LOGGER.error("Error trying to get map of addresses (IP and Port) " +
+                    "of the compute instance for order: " + order, e);
+        }
+    }
+
+    public boolean isActiveConnectionFromInstance(ComputeOrderInstance computeOrderInstance) {
```

Comment:
I think `isInstanceReachable` would be a better name for this method. Also, I think this method could be summarized to `return sshConnectivity.checkSSHConnectivity(computeOrderInstance)` or something like that.

Reply:
Done!